frontend/src/concepts/pipelines/context/PipelinesContext.tsx

```diff
@@ -50,14 +50,17 @@ const PipelinesContext = React.createContext<PipelineContext>({
   serverTimedOut: false,
   ignoreTimedOut: () => undefined,
   namespace: '',
+  // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
   project: null as unknown as ProjectKind,
   refreshState: async () => undefined,
   refreshAPIState: () => undefined,
   getJobInformation: () => ({
     loading: false,
     data: null,
   }),
+  // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
   apiState: { apiAvailable: false, api: null as unknown as PipelineAPIState['api'] },
+  // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
   metadataStoreServiceClient: null as unknown as MetadataStoreServicePromiseClient,
```

Comment:
Agreed, I don't think we care about outside of Context -- that's a dev problem, and it would be pretty evident and fixed before code ever hit runtime.

🤔 So... we could change this to `{} as Type` but my concern is we have code looking at some value being a truthy value -- like in a loading state internal to the context component.

I don't see that as a high chance, but it is a non-zero chance. Perhaps ignoring this `as` usage and coming back with a Tech Debt is the better way to go. In the end this PR is not going to get super great at test coverage due to the wide reaching changes. Hopefully Cypress & Jest play ball and catch all the corners, but we are not 100% test covered. 

@christianvogt does that sound reasonable?

Reply:
Some code reaching for the context today without the provider is very likely to break.

We are always render the providers where we expect child components to depend on them. The use case you bring up would be very edge and likely overlooked by the dev who would have implemented such a case. ie to render a loading state that depends on the context in which the loading state is rendered within thus using the default value. <_snake eats itself_>

@andrewballantyne I think it's fair that we tackle these contexts in a separate PR instead of along with the rest.